### ISSUE TITLE
How to properly logout across all devices?

### ISSUE THREAD

**Issue by RPM1984** [IdentityServer/IdentityServer3#2547]
Consider the scenario if i have a web app using implicit flow and a mobile (iOS) app using code flow (so both browser-based where the cookie/session is at server).

Let's say a user is logged into both platforms. Then they logout of the web app (using the `/endsession`) endpoint, how will the iOS app know that the user has ended their session at the server? From the iOS apps point of view, then still have an identity token/access token they received/stored earlier, and the user is logged in locally.

I know in the SPA-Implicit sample app, the web does a "ping" in the background using an `<iframe>` to check the session. How should the iOS app do this? A hidden web view that does the same thing?

Basically, my question is what is the _correct_ way to logout across all devices :smile: 

Thanks in advance.

**Comment by brockallen** [comment]
> Consider the scenario if i have a web app using implicit flow and a mobile (iOS) app using code flow (so both browser-based where the cookie/session is at server).

They certainly are not sharing the same browser, thus the concept of single signout does not apply (at least in the context of the browser).

> Let's say a user is logged into both platforms. Then they logout of the web app (using the /endsession) endpoint, how will the iOS app know that the user has ended their session at the server?

It wouldn't.

> How should the iOS app do this?

Should it?

> Basically, my question is what is the correct way to logout across all devices 

Not sure there is a correct way. Do you have a specific requirement for this?

**Comment by RPM1984** [comment]
@brockallen you're right.. /facepalm

I don't have a specific requirement, but the question arose when thinking about whether the iOS app should even bother doing the `/endsession` call.

Since the iOS app handles the login session locally (e.g once it gets back the identity/access token, it logs the user in locally), what advantage would there be of even doing the logout? All that would do is show the login screen when they try and login again, whereas _not_ doing the endsession call and trying to login again would skip the login screen, assuming the identity session is still valid.

So it sounds like a non browser-based client should not actually bother logging out?

**Comment by brockallen** [comment]
> I don't have a specific requirement, but the question arose when thinking about whether the iOS app should even bother doing the /endsession call.

Only if it's using the system browser, which I doubt it is.

**Comment by RPM1984** [comment]
@brockallen currently the iOS app is opening an embedded web view to do the login. Is that what you mean?

**Comment by brockallen** [comment]
No, that's an embedded browser, not the system browser. Embedded is not shared with other app, whereas the system one is.

**Comment by RPM1984** [comment]
@brockallen right - so you're basically saying it's only worth doing the logout if i had multiple apps on the same device that use the system browser, since that's the only way to do SSO/SSO logout in that case...

**Comment by brockallen** [comment]
Yes.

**Comment by RPM1984** [comment]
Thanks.
